src/server/accounting.c

```diff
@@ -2270,19 +2280,33 @@ void log_alter_records_for_attrs(job *pjob, svrattrl *plist) {
 								fmt = "%s=%s";
 							else
 								fmt = "%s=\"%s\"";
-							pbs_asprintf(&logstr, fmt, cur_svr->al_name, cur_svr->al_value);
+							pbs_asprintf(&per_alter_buf, fmt, cur_svr->al_name, cur_svr->al_value);
 							break;
 						}
 					}
 				}
-				if (logstr == NULL && cur_plist->al_value[0] == '\0') { /* unset */
-					pbs_asprintf(&logstr, "%s%s%s=UNSET", cur_plist->al_name, cur_plist->al_resc ? "." : "", cur_plist->al_resc ? cur_plist->al_resc : "");
+				if (per_alter_buf == NULL && cur_plist->al_value[0] == '\0') { /* unset */
+					pbs_asprintf(&per_alter_buf, "%s%s%s=UNSET", cur_plist->al_name, cur_plist->al_resc ? "." : "", cur_plist->al_resc ? cur_plist->al_resc : "");
+				};
+
+				if (entire_record[0] != '\0')
+					if (pbs_strcat(&entire_record, &entire_record_len, " ") == NULL)
+						error = 1;
+
+				if (error == 0)
+					if (pbs_strcat(&entire_record, &entire_record_len, per_alter_buf) == NULL)
+						error = 1;
+
+				free(per_alter_buf);
+				per_alter_buf = NULL;
+				if (error) {
+					free(entire_record);
+					return;
 				}
-				account_record(PBS_ACCT_ALTER, pjob, logstr);
-				free(logstr);
-				logstr = NULL;
 			}
 			free_svrattrl(svrattrl_list);
 		}
 	}
+	account_record(PBS_ACCT_ALTER, pjob, entire_record);
```

Comment:
do you wanna check if entire_record[0] == '\0'  before calling account_record() ?